components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -1192,4 +1231,52 @@ private void provision(ProvisioningOperation provisioningMethod, SCIMObject prov
             throw new CharonException("Error while provisioning to externaluser store in dumb mode.", e);
         }
     }
+
+    private String getUserStoreDomainFromSP() throws IdentityApplicationManagementException {
+
+        ThreadLocalProvisioningServiceProvider threadLocalSP = IdentityApplicationManagementUtil
+                .getThreadLocalProvisioningServiceProvider();
+        ServiceProvider serviceProvider = null;
+        if (threadLocalSP.getServiceProviderType() == ProvisioningServiceProviderType.OAUTH) {
+            serviceProvider = ApplicationManagementService.getInstance()
+                                                          .getServiceProviderByClientId(
+                                                                  threadLocalSP.getServiceProviderName(),
+                                                                  "oauth2", threadLocalSP.getTenantDomain());
+        } else {
+            serviceProvider = ApplicationManagementService.getInstance().getServiceProvider(
+                    threadLocalSP.getServiceProviderName(), threadLocalSP.getTenantDomain());
+        }
+
+        if (serviceProvider != null && serviceProvider.getInboundProvisioningConfig() != null) {
+            return serviceProvider.getInboundProvisioningConfig().getProvisioningUserStore();
+        }
+        return null;
+    }
+
+    private Group addDomainToUserMembers(Group group, String userStoreDomain) throws CharonException {
+        List<String> membersId = group.getMembers();
+
+        if (StringUtils.isBlank(userStoreDomain) || membersId == null || membersId.isEmpty()) {
+            return group;
+        }
+
```

Comment:
Better to create a map with ID and displayname initially instead using two independent arryalists